tests/cpp_methods/unittest_cpp_methods.cc

```diff
@@ -48,31 +56,392 @@ class filter_basic: public tensor_filter_cpp {
+    g_usleep(100000);
+
+    gst_object_unref (pipeline);
+
+    EXPECT_EQ (filter_basic::resultCompare (tmp2, tmp1), 0);
+  }
+  g_free (str_pipeline);
+
+  g_free (tmp1);
+  g_free (tmp2);
+  EXPECT_EQ (basic._unregister(), 0);
+}
+
+/** @brief Negative case for the simpliest execution path */
+TEST (cpp_filter_on_demand, unregistered_01_n)
+{
+  filter_basic basic("basic_01");
+  gchar *str_pipeline = g_strdup_printf
+      ("videotestsrc num-buffers=3 ! videoconvert ! videoscale ! video/x-raw,width=4,height=4,format=RGB ! tensor_converter ! tensor_filter framework=cpp model=XXbasic_01 ! fakesink");
+  GError *err = NULL;
+  GstElement *pipeline = gst_parse_launch (str_pipeline, &err);
+
+  EXPECT_NE (pipeline, nullptr);
+
+  EXPECT_EQ (basic._register(), 0);
+  gst_object_unref(pipeline);
+  pipeline = gst_parse_launch (str_pipeline, NULL);
+  EXPECT_NE (pipeline, nullptr);
+  if (pipeline)
+    gst_object_unref(pipeline);
+  EXPECT_EQ (basic._unregister(), 0);
+
+  basic._unregister();
+  g_free (str_pipeline);
+  EXPECT_NE (basic._unregister(), 0);
+}
+
+TEST (cpp_filter_obj, base_01_n)
+{
+  char *tmp1 = getTempFilename ();
+  char *tmp2 = getTempFilename ();
+  char *tmp3 = getTempFilename ();
+
+  EXPECT_NE (tmp1, nullptr);
+  EXPECT_NE (tmp2, nullptr);
+  EXPECT_NE (tmp3, nullptr);
+  gchar *str_pipeline = g_strdup_printf
+      ("videotestsrc num-buffers=5 ! videoconvert ! videoscale ! video/x-raw,width=4,height=4,format=RGB ! tensor_converter ! tee name=t t. ! queue name=q1 ! tensor_filter framework=cpp model=basic_so_01,%slibcppfilter_test.so ! filesink location=%s t. ! queue name=q2 ! filesink location=%s t. ! queue ! tensor_filter framework=cpp model=basic_so_03,%slibcppfilter_test.so ! filesink location=%s", path_to_lib, tmp1, tmp2, path_to_lib, tmp3);
+GError *err = NULL;
+  GstElement *pipeline = gst_parse_launch (str_pipeline, &err);
+
+  EXPECT_NE (pipeline, nullptr);
+
+  if (pipeline) {
+    EXPECT_NE (setPipelineStateSync (pipeline, GST_STATE_PLAYING, 500000U), 0);
+
+    gst_object_unref (pipeline);
+
+  }
+  g_free (str_pipeline);
+
+  g_free (tmp1);
+  g_free (tmp2);
+  g_free (tmp3);
+}
+
+TEST (cpp_filter_obj, base_02_n)
+{
+  char *tmp1 = getTempFilename ();
+  char *tmp2 = getTempFilename ();
+  char *tmp3 = getTempFilename ();
+
+  EXPECT_NE (tmp1, nullptr);
+  EXPECT_NE (tmp2, nullptr);
+  EXPECT_NE (tmp3, nullptr);
+  gchar *str_pipeline = g_strdup_printf
+      ("videotestsrc num-buffers=5 ! videoconvert ! videoscale ! video/x-raw,width=4,height=4,format=RGB ! tensor_converter ! tee name=t t. ! queue name=q1 ! tensor_filter framework=cpp model=basic_so_01,%slibcppfilter_test.so ! filesink location=%s t. ! queue name=q2 ! filesink location=%s t. ! queue ! tensor_filter framework=cpp model=basic_so_03,%slibcppfilter_test.so ! filesink location=%s", path_to_lib, tmp1, tmp2, path_to_lib, tmp3);
+
+  GError *err = NULL;
+  GstElement *pipeline = gst_parse_launch (str_pipeline, &err);
```

Comment:
clear err if failed (or set null, I think err is unnecessary in testcase)

Reply:
Fixed.